src/test/java/org/sonarsource/dotnet/shared/sarif/SarifParser01And04Test.java

```
@@ -90,4 +89,17 @@ public void sarif_version_0_4() throws IOException {
     verify(callback, Mockito.never()).onProjectIssue(Mockito.anyString(), Mockito.anyString());
   }
 
+  @Test
+  public void sarif_version_0_4_file_level() throws IOException {
+    SarifParserCallback callback = mock(SarifParserCallback.class);
+    new SarifParser01And04(getRoot("v0_4_file_level_issue.json")).parse(callback);
+
+    InOrder inOrder = inOrder(callback);
+    inOrder.verify(callback).onFileIssue(eq("S104"), Mockito.anyString(), eq("Dummy"));
+    inOrder.verify(callback).onFileIssue(eq("S105"), Mockito.anyString(), eq("Dummy"));
+    inOrder.verify(callback).onIssue(eq("S105"), Mockito.anyString(), eq("Dummy"), eq(1), eq(0), eq(1), eq(1));
+    inOrder.verify(callback).onIssue(eq("S106"), Mockito.anyString(), eq("Dummy"), eq(1), eq(0), eq(1), eq(0));
```

So I would change that to fail fast (see my other comment).